exla/lib/exla/defn.ex

```diff
@@ -300,9 +300,9 @@ defmodule EXLA.Defn do
     EXLA.Op.get_tuple_element(op, index)
   end
 
-  defp to_operator(:dot, [left, axes1, right, axes2], %{type: type}, state) do
+  defp to_operator(:dot, [left, contract_axes1, _batch_axes1, right, contract_axes2, _batch_axes2], %{type: type}, state) do
     precision = state.precision
-    EXLA.Op.dot_general(to_type(left, type), to_type(right, type), {axes1, axes2}, precision)
+    EXLA.Op.dot_general(to_type(left, type), to_type(right, type), {contract_axes1, contract_axes2}, precision)
```

Comment:
It turns out that EXLA already impĺemented dot through dot_general. 

In exla/c_src/exla/exla_nif_util.cc there is the `get_dot_dimension_numbers` function in which we need to deal with receiving a tuple of tuples instead of a tuple of axes lists. I think that changing this, everything should work.

However, I don't yet know what should be changed in there for this to work